provider/common/bootstrap.go

```diff
@@ -214,6 +245,38 @@ func BootstrapInstance(ctx environs.BootstrapContext, env environs.Environ, args
 	return result, selectedSeries, finalize, nil
 }
 
+func startInstanceZones(env environs.Environ, args environs.StartInstanceParams) ([]string, error) {
+	zonedEnviron, ok := env.(ZonedEnviron)
+	if !ok {
+		return nil, errors.NotImplementedf("ZonedEnviron")
+	}
+
+	// Attempt creating the instance in each of the availability
+	// zones, unless the args imply a specific zone.
+	zone, err := zonedEnviron.DeriveAvailabilityZone(args)
```

Comment:
shouldn't Derive be allowed to say there are potentially a couple of zones that might work? (I'm thinking things like spaces where you define a space that does span zones, but it might not span all possible zones.)

Similarly, we've talked about being able to do application level constraints where you specify a subset of the zones.

Something like:
 juju deploy postgres --constraints zones=us-east-1a,us-east-1b,us-east-1c

This is more relevant on MAAS where zones are *not* equivalent. Some of that is abuse of the Zone/Region terminology, but it is a matter of course that people do things like "create a zone grouping some machines together for a specific purpose", and they want to be able to include a range.

Reply:
Maybe later. Right now this is what we need.